src/View/Widgets/AppMenu.vala

```diff
@@ -139,6 +157,37 @@ public class Files.AppMenu : Gtk.Popover {
         Files.icon_view_settings.changed["zoom-level"].connect (on_zoom_setting_changed);
         Files.list_view_settings.changed["zoom-level"].connect (on_zoom_setting_changed);
         Files.column_view_settings.changed["zoom-level"].connect (on_zoom_setting_changed);
+
+        // Initialize and connect dateformat buttons
+        switch (app_settings.get_enum ("date-format")) {
+            case DateFormatMode.ISO:
+                iso_button.active = true;
+                break;
+            case DateFormatMode.LOCALE:
+                locale_button.active = true;
+                break;
+            case DateFormatMode.INFORMAL:
+                informal_button.active= true;
+                break;
+            default:
+                assert_not_reached ();
+        }
+
+        // Using a local function to avoid making radiobuttons class members
+        void on_dateformat_changed () {
+            if (iso_button.active) {
+                app_settings.set_enum ("date-format", DateFormatMode.ISO);
+            } else if (locale_button.active) {
+                app_settings.set_enum ("date-format", DateFormatMode.LOCALE);
+            } else {
+                app_settings.set_enum ("date-format", DateFormatMode.INFORMAL);
+            }
+        }
+
+        iso_button.toggled.connect (() => on_dateformat_changed ());
+        locale_button.toggled.connect (() => on_dateformat_changed () );
+        informal_button.toggled.connect (() => on_dateformat_changed ());
+
```

Comment:
I think the following code would be better rather than creating an local method:

```vala
        iso_button.toggled.connect (() => {
            if (iso_button.active) {
                app_settings.set_enum ("date-format", DateFormatMode.ISO);
            }
        });
        locale_button.toggled.connect (() => {
            if (locale_button.active) {
                app_settings.set_enum ("date-format", DateFormatMode.LOCALE);
            }
        });
        informal_button.toggled.connect (() => {
            if (informal_button.active) {
                app_settings.set_enum ("date-format", DateFormatMode.INFORMAL);
            }
        });
```